### PR TITLE
fix: #848 user name disappearing when log out

### PR DESCRIPTION
**Pull request by J0taFerreira**
- fixed user name disappearing when log out was clicked
- button now have loading state
- button refactored to use our reusable button component

### CODE REVIEW

File: frontend/src/components/common/Button.vue [hunk 1]
```diff
@@ -1,16 +1,24 @@
-<script setup>
+<script setup lang="ts">
 import Button from 'primevue/button';
+import type { PropType } from 'vue';
+import { IconPosition } from '../../enum/IconEnum';
```
Comment (ianliuwk1019) [inline frontend/src/components/common/Button.vue:4]: Use better reference than "../..".

File: frontend/src/components/common/Button.vue [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 });
 </script>
 <template>
-    <Button class="nr-button"
-        ><span>{{ props.label }}</span> <slot class="icon"></slot
-    ></Button>
+    <Button class="nr-button">
+        <slot class="icon" v-if="iconPos === IconPosition.left"></slot>
+        <span>{{ props.label }}</span>
+        <slot class="icon" v-if="iconPos === IconPosition.right"></slot>
```
Comment (ianliuwk1019) [inline frontend/src/components/common/Button.vue:20]: Can "v-else" work?

File: frontend/src/components/common/Button.vue [hunk 3]
```diff
@@ -1,16 +1,24 @@
...
+import { IconPosition } from '../../enum/IconEnum';
 const props = defineProps({
     label: {
         type: String,
         default: 'Click',
     },
+    iconPos: {
+        type: String as PropType<IconPosition>,
+        default: IconPosition.right,
+    },
```
Comment (MCatherine1994) [inline frontend/src/components/common/Button.vue:13]: I'm not so sure why we need this iconPos property?

File: frontend/src/components/common/Button.vue (hunk 2 above)
Comment (MCatherine1994) [inline frontend/src/components/common/Button.vue:20]: not so sure why we need these slot element?

File: frontend/src/components/common/Button.vue (hunk 3 above)
Comment (J0taFerreira) [reply frontend/src/components/common/Button.vue:13]: Hello, Catherine! The sign out button has the icon on a before the text, to use this component I did this so we could just pass the side we want the icon

File: frontend/src/components/common/ProfileSidebar.vue [hunk 4]
```diff
@@ -121,14 +136,25 @@ import { IconSize } from '@/enum/IconEnum';
...
-    color: #000;
+    background-color: transparent !important;
+    color: $text-secondary !important;
     cursor: pointer;
 }
 
 .sign-out:hover {
     background-color: transparent;
     background-color: #ffffff;
+    color-scheme: $text-secondary !important;
```
Comment (ianliuwk1019) [inline frontend/src/components/common/ProfileSidebar.vue:147]: I am not sure why this is "color-scheme", did you mean to use "color"?
Without this added, does that work?

File: frontend/src/components/common/ProfileSidebar.vue [hunk 5]
```diff
@@ -121,14 +136,25 @@ import { IconSize } from '@/enum/IconEnum';
...
+}
+
+.sign-out:focus {
+    background-color: transparent;
+    background-color: #ffffff;
+    color-scheme: $text-secondary !important;
```
Comment (ianliuwk1019) [inline frontend/src/components/common/ProfileSidebar.vue:153]: Same comment above.

File: frontend/src/components/common/Button.vue (hunk 2 above)
Comment (J0taFerreira) [reply frontend/src/components/common/Button.vue:20]: the slot is the icon, if the button receive left as props, the first slot (icon on will be before the text)  will be the one visible, else the second slot (icon will be after the text) will be the one visible.

File: frontend/src/components/common/Button.vue (hunk 3 above)
Comment (MCatherine1994) [reply frontend/src/components/common/Button.vue:13]: Oh, I got why we need it, for the icon

File: frontend/src/components/common/Button.vue (hunk 2 above)
Comment (MCatherine1994) [reply frontend/src/components/common/Button.vue:20]: yes, got it, thanks for the explanation

**Review by MCatherine1994 — DISMISSED** [review]

**Review by ianliuwk1019 — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/Button.vue (hunk 2 above)
Comment (J0taFerreira) [reply frontend/src/components/common/Button.vue:20]: I tried to use a v-else but it gave the " v-else/v-else-if has no adjacent v-if or v-else-if " typescript com·plaint.

File: frontend/src/components/common/ProfileSidebar.vue (hunk 4 above)
Comment (J0taFerreira) [reply frontend/src/components/common/ProfileSidebar.vue:147]: it works the same but i'll use color to be more clear.

File: frontend/src/components/common/ProfileSidebar.vue (hunk 5 above)
Comment (J0taFerreira) [reply frontend/src/components/common/ProfileSidebar.vue]: same comment as above

File: frontend/src/components/common/Button.vue (hunk 2 above)
Comment (ianliuwk1019) [reply frontend/src/components/common/Button.vue:20]: Hmm... a bit surprised vue does not have this support.
We can only leave it as v-if then.

**Review by ianliuwk1019 — APPROVED** [review]

**Pull request merged**